Require stream destroy without fix

danieljbruce opened the pull request:
This PR illustrates a test that passes if the chunks get sent back whole so that we can more easily explore solutions that will work and solutions that don't.

danieljbruce committed:
A unit test for passing data through

Wrote a unit test here for passing data through the transformer that works.

# Conflicts:
#	system-test/bigtable.ts

danieljbruce committed:
Add a chunk here and use it in the stream

Push the chunk in the stream

danieljbruce committed:
Add more chunks to the unit test

There should be 209 chunks of data per stream push to simulate normal behavior

danieljbruce committed:
Works and reports a row count

Updated the test again so it works and indeed reports a row count.

danieljbruce committed:
Modify the code and get uneven amount of chunks

Push chunks until stream ends

danieljbruce committed:
Finalize unit test that sends whole chunks

This test passes when whole chunks get processed and make it to the user while a stream ends spontaneously.

danieljbruce committed:
Merge branch 'main' of https://github.com/googleapis/nodejs-bigtable into require-stream-destroy-without-fix

# Conflicts:
#	system-test/bigtable.ts

alexander-fenster commented on system-test/bigtable.ts:
```diff
@@ -488,6 +491,83 @@ describe('Bigtable', () => {
+        };
+        stream.push(data);
+      }
+      setTimeout(() => {
```
Change this to `process.nextTick`.

alexander-fenster replied:
Actually, maybe even emit each chunk at `nextTick` and then `end` at `nextTick` as well.

danieljbruce replied:
I gave your former suggestion a try. For your latter suggestion do you mean something like this?

```
let chunksPushed = 0;
process.nextTick(() => {
    if (chunksPushed > 83) {
        pushChunks(chunkSize);
        chunksPushed++;
    } else {
        stream.emit('end');
    }
});
```

danieljbruce committed:
Change the setTimeout to nextTick

Change this setTimeout to next tick so that the test behaves as intended no matter what the runtime of the environment is.
